var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -233,7 +237,8 @@ def cmake_args(self):
                 define(project_name_base + "_ENABLE_SEACAS", True),
             ]
         )
-
+        if "~shared" in self.spec:
+            options.append(self.define("Seacas_EXTRA_LINK_FLAGS", "z;dl"))
```

Comment:
Ideally these should be coming from the non-shared builds of netCDF and HDF5 and not needed here, but that does not seem to be working so we specify them here also.

Reply:
Seems reasonable for a work around. Do you think something is broken? i.e. Is this an ideally, b/c it works with a build external to spack, or a ideally as in it would be nice?

Reply:
This is done the same way in the SEACAS build system through a configure script right now because netCDF and HDF5 need it there as well. It does not seem to work without them external to Spack either. Ideally, it would be handled in the build systems of netCDF and HDF5.

Reply:
`Ideally` as in I think that the dependency chain from HDF5 (built in spack) is not being carried through to the applications that eventually use HDF5.  we are adding the libz, libdl dependency in seacas, but the dependency is at the hdf5 level.  Not sure where the missing link is, but it would be ideal if the dependency were specified at the level of the library that has the dependency and then is carried through.  I don't know if this is an error in the HDF5 CMake, or if there is something we are doing in our CMake that is losing the library dependency. 

For now, this is probably the best solution.